sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -638,6 +638,9 @@ def try_split(self, bundle_split_request):
                       last_primary_element=primary_end,
                       first_residual_element=residual_start)])
 
+    if not split_response.channel_splits:
+      raise ValueError('Unable to find an effective split.')
```

Comment:
Just curious why we treat empty split as a value error? From my understanding, the split should be somehow best-effort, not a guaranteed request.